cachi2/core/package_managers/pip.py

```diff
@@ -204,20 +200,28 @@ def _get_pip_metadata(package_dir: RootedPath) -> tuple[str, str]:
         name = name or setup_cfg.get_name()
         version = version or setup_cfg.get_version()
 
-    if name:
-        log.info("Resolved package name: %r", name)
+    if not name:
+        log.info("Processing metadata from git repository")
+        try:
+            repo_path = get_repo_id(package_dir.root).parsed_origin_url.path
+            package_subpath = package_dir.subpath_from_root
+            resolved_path = Path(repo_path).joinpath(package_subpath)
+            normalized_path = pkg_resources.safe_name(str(resolved_path))
+            name = normalized_path.replace(".git", "")
```

Comment:
Any idea how to handle leading and trailing "-" ?
Is there a better way instead of `strip()` ?

Reply:
Good point. I think `strip()` is fine (and maybe we should strip `.` as well)

```python
In [2]: pkg_resources.safe_name(".foo")
Out[2]: '.foo'
```